ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/caching/Weight.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.caching
+
+import com.daml.ledger.participant.state.kvutils.Bytes
+import com.github.benmanes.caffeine.{cache => caffeine}
+import com.google.protobuf.MessageLite
+
+trait Weight[-T] {
+  def weigh(value: T): Size
+}
+
+object Weight {
+  def apply[T](implicit weight: Weight[T]): Weight[T] =
+    weight
+
+  def weigh[T](value: T)(implicit weight: Weight[T]): Size =
+    weight.weigh(value)
+
+  def weigher[Key: Weight, Value: Weight]: caffeine.Weigher[Key, Value] =
+    new WeightWeigher[Key, Value]
+
+  def ofCache[Key, Value](cache: Cache[Key, Value])(
+      implicit keyWeight: Weight[Key],
+      valueWeight: Weight[Value],
+  ): Size =
+    cache.entries.map { case (key, value) => keyWeight.weigh(key) + valueWeight.weigh(value) }.sum
```

Comment:
You can use `cache.policy().eviction()` to obtain the `weightedSize()`

Reply:
Assuming the same weigher, of course.

Reply:
Thanks! This let me delete a chunk of code. 😀

I opened #5461 with your suggestion.